src/core/src/p256.rs

```diff
@@ -0,0 +1,90 @@
+use hd_keys_curves_wasm::HDDeriver;
+use p256::elliptic_curve::sec1::{FromEncodedPoint, ToEncodedPoint};
+
+use crate::utils::read_4_byte_be;
+
+pub fn hd_key_derive(data: Vec<u8>) -> Result<Vec<u8>, Vec<u8>> {
+    // 1st arg is a byte for the curve type, 0 is Nist Prime256, 1 is secp256k1
+    // 2nd arg is a 4 byte big-endian integer for the number of bytes in id
+    // 3rd arg is the byte sequence for id
+    // 4th arg is a 4 byte big-endian integer for the number of bytes in cxt
+    // 5th arg is the byte sequence for cxt
+    // 6th arg is a 4 byte big-endian integer for the number of root keys
+    // 7th arg is a variable number of root keys each 64 bytes in length
+    let mut offset = 1;
+
+    let id_len = read_4_byte_be(&data, &mut offset)? as usize;
+    let id = &data[offset..offset + id_len];
+    offset += id_len;
+
+    let cxt_len = read_4_byte_be(&data, &mut offset)? as usize;
+    let cxt = &data[offset..offset + cxt_len];
+    offset += cxt_len;
+
+    let root_keys_len = read_4_byte_be(&data, &mut offset)? as usize;
+    let buffer = &data[offset..];
+
+    if root_keys_len < 2 || (root_keys_len * 33) > buffer.len() {
```

Comment:
Same comments and k256

Reply:
not a meaningful gas difference to do this, no worth the changes to working contracts IMO